L/LLVM@8/build_tarballs.jl

```diff
@@ -0,0 +1,280 @@
+# `ld -v`, which is hilariously wrong.
+CMAKE_FLAGS+=(-DLLVM_HOST_TRIPLE=${target})
+
+# Tell LLVM which compiler target to use, because it loses track for some reason
+CMAKE_FLAGS+=(-DCMAKE_C_COMPILER_TARGET=${target})
+CMAKE_FLAGS+=(-DCMAKE_CXX_COMPILER_TARGET=${target})
+CMAKE_FLAGS+=(-DCMAKE_ASM_COMPILER_TARGET=${target})
+
+if [[ "${target}" == *apple* ]]; then
+    # On OSX, we need to override LLVM's looking around for our SDK
+    CMAKE_FLAGS+=(-DDARWIN_macosx_CACHED_SYSROOT:STRING=/opt/${target}/${target}/sys-root)
+
+    # LLVM actually won't build against 10.8, so we bump ourselves up slightly to 10.9
+    export MACOSX_DEPLOYMENT_TARGET=10.9
+    export LDFLAGS=-mmacosx-version-min=10.9
+
+    # We need to link against libc++ on OSX
+    CMAKE_FLAGS+=(-DLLVM_ENABLE_LIBCXX=ON)
+fi
+
+if [[ "${target}" == *apple* ]] || [[ "${target}" == *freebsd* ]]; then
+    # On clang-based platforms we need to override the check for ffs because it doesn't work with `clang`.
+    export ac_cv_have_decl___builtin_ffs=yes
+
+    # We don't use X-ray on BSD systems
+    CMAKE_FLAGS+=(-DCOMPILER_RT_BUILD_XRAY=OFF)
+fi
+
+if [[ "${target}" == *mingw* ]]; then
+    CMAKE_CPP_FLAGS="${CMAKE_CPP_FLAGS} -remap -D__USING_SJLJ_EXCEPTIONS__ -D__CRT__NO_INLINE"
+    # Windows is case-insensitive and some dependencies take full advantage of that
+    echo "BaseTsd.h basetsd.h" >> /opt/${target}/${target}/include/header.gcc
+fi
+
+if [[ "${target}" == *musl* ]]; then
+    # Taken from https://git.alpinelinux.org/cgit/aports/tree/main/compiler-rt/APKBUILD
+    CMAKE_FLAGS+=(-DCOMPILER_RT_INCLUDE_TESTS=ON)
+    CMAKE_FLAGS+=(-DCOMPILER_RT_BUILD_SANITIZERS=OFF)
+    CMAKE_FLAGS+=(-DCOMPILER_RT_BUILD_XRAY=OFF)
+fi
+
+if [[ "${target}" == *freebsd* ]]; then
+    # On FreeBSD, we must force even statically-linked code to have -fPIC
+    CMAKE_FLAGS+=(-DCMAKE_POSITION_INDEPENDENT_CODE=TRUE)
+fi
+
+# Build!
+cmake -GNinja ${LLVM_SRCDIR} ${CMAKE_FLAGS[@]}
+cmake -LA || true
+ninja -j${nproc}
+
+# Install!
+ninja install -j${nproc}
+
+# move clang products out of $prefix/bin to $prefix/tools
+mv ${prefix}/bin/clang* ${prefix}/tools/
+mv ${prefix}/bin/scan-* ${prefix}/tools/
+mv ${prefix}/bin/c-index* ${prefix}/tools/
+mv ${prefix}/bin/git-clang* ${prefix}/tools/
+mv ${prefix}/bin/lld* ${prefix}/tools/
+
+# Life is harsh on Windows and dynamic libraries are
+# expected to live alongside the binaries. So we have
+# to copy the *.dll from bin/ to tools/ as well...
+if [[ "${target}" == *mingw* ]]; then
+    cp ${prefix}/bin/*.dll ${prefix}/tools/
+fi
+
+# Work around llvm-config bug by creating versioned symlink to libLLVM
+# https://github.com/JuliaLang/julia/pull/30033
+if [[ "${target}" == *darwin* ]]; then
+    LLVM_VER=$(basename $(echo ${prefix}/tools/clang-*.*))
+    ln -s libLLVM.dylib ${prefix}/lib/libLLVM-${LLVM_VER##*-}.dylib
+fi
+
+# Lit is a python dependency and there is no proper install target
+cp -r ${LLVM_SRCDIR}/utils/lit ${prefix}/tools/
+"""
```

Comment:
Should we install the license file?  It's currently missing and audit is complaining.  This command should do the job:
```sh
install_license ${WORKSPACE}/srcdir/llvm-project/llvm/LICENSE.TXT
```